cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -1415,16 +1455,36 @@ func generateMountsModeRecover(mst *initramfsMountsState) error {
 	// ubuntu-data, and as such we can proceed with copying files from there
 	// onto the tmpfs
 	// Proceed only if we trust ubuntu-data to be paired with ubuntu-save
+	hybrid := model.Classic() && model.KernelSnap() != nil
+
 	if machine.trustData() {
-		// TODO: erroring here should fallback to copySafeDefaultData and
-		// proceed on with degraded mode anyways
-		if err := copyUbuntuDataAuth(boot.InitramfsHostUbuntuDataDir, boot.InitramfsDataDir); err != nil {
-			return err
+		hostSystemData := boot.InitramfsHostWritableDir(model)
+		recoverySystemData := boot.InitramfsWritableDir(model, false)
+		if hybrid {
+			if err := importHybridUserData(
+				hostSystemData,
+				filepath.Join(boot.InitramfsRunMntDir, "base"),
```

Comment:
Note that we want to mount now the base directly on `/sysroot`, so this needs a TODO so we remember to change this. See https://github.com/canonical/snapd/pull/14537